File: exam10/app/src/main/java/com/example/exam10/MainActivity.java
package com.example.exam10;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    // Activity Result Launcher 정의
    private final ActivityResultLauncher<Intent> launcher = registerForActivityResult(
            new ActivityResultContracts.StartActivityForResult(),
            result -> {
                if (result.getResultCode() == Activity.RESULT_OK) {
                    // SecondActivity에서 전달된 데이터를 가져옴
                    Intent data = result.getData();
                    if (data != null) {
                        String returnedData = data.getStringExtra("returnedData");
                        // 데이터를 사용하는 코드 추가 (예: 텍스트뷰에 표시)
//                        TextView textView = findViewById(R.id.textView);
//                        textView.setText(returnedData);

                        EditText edName = findViewById(R.id.edName);
                        edName.setText(returnedData);
                    }
                }
            }
    );

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        findViewById(R.id.btnGotoSecond).setOnClickListener(v -> {
            Intent intent = new Intent(this, SecondActivity.class);

            EditText edName = findViewById(R.id.edName);

            intent.putExtra("name", edName.getText().toString());
            launcher.launch(intent);
        });

        findViewById(R.id.btnMap).setOnClickListener(v -> {
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("geo:37.5662952,126.9779451"));

            startActivity(intent);
        });
        }



}

File: Ex1002/app/src/main/java/com/gbox3d/ex1002/MainActivity.kt
package com.gbox3d.ex1002

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.activity.result.contract.ActivityResultContracts

class MainActivity : AppCompatActivity() {

    // ActivityResultLauncher 초기화
    private val startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
        if (result.resultCode == Activity.RESULT_OK) {
            val intent = result.data
            // 결과 처리
            val message = intent?.getStringExtra("msg")
            Log.d("myLog", "Received message: $message")
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btnGoSecond).setOnClickListener {
            var intent = Intent(this,SecondActivity::class.java)

            Log.d("myLog","send Request")

            intent.putExtra("msg","Doromith")


            startForResult.launch(intent)
        }

    }

}